admin-dev/themes/new-theme/js/pages/product/components/generator/AttributesSelector.vue

```diff
@@ -169,42 +195,49 @@
       },
     },
     methods: {
-      initDataSetConfig() {
+      initDataSetConfig(): void {
         const searchItems = this.getSearchableAttributes();
+
+        const letters = [
+          'name',
+          'value',
+          'group_name',
+        ];
+
         this.searchSource = new Bloodhound({
-          datumTokenizer: Tokenizers.obj.letters(
-            'name',
-            'value',
-            'group_name',
-          ),
+          datumTokenizer: Tokenizers.obj.letters(letters),
           queryTokenizer: Bloodhound.tokenizers.nonword,
           local: searchItems,
         });
 
         this.dataSetConfig = {
           source: this.searchSource,
-          display: (item) => `${item.group_name}: ${item.name}`,
+          display: (item: Attribute) => `${item.group_name}: ${item.name}`,
           value: 'name',
           minLength: 1,
-          onSelect: (attribute, e, $searchInput) => {
-            const attributeGroup = {
+          onSelect: (attribute: Attribute, e: Record<string, any>, $searchInput: JQuery) => {
```

Comment:
```suggestion
          onSelect: (attribute: Attribute, e: JQueryEventObject, $searchInput: JQuery) => {
```

We use this event type in the autocomplete component, should we make them match? Although it could probably be replaced by `JQuery.ChangeEvent` or similar since `JQueryEventObject` is deprecated

Reply:
Changed to `JQueryEventObject`, you're right it's a miss